src/rgw/rgw_op.cc

```diff
@@ -5013,6 +5013,13 @@ void RGWDeleteObj::execute(optional_yield y)
         etag = astate->attrset[RGW_ATTR_ETAG].to_str();
       }
 
+      // if the object does not exist, it is not allowed to create a delete_marker
+      if(s->bucket->get_info().versioning_enabled()&&!s->object->have_instance()){
```

Comment:
you need white-space around operators && and ||, and later, after the arguments to a function.

Reply:
oops, thanks for pointing out. I'll update it after I fix the error for make check